src/tqec/sketchup/block_graph.py

```diff
@@ -551,14 +580,117 @@ def display(
         pipe_length: float = 2.0,
     ) -> ColladaDisplayHelper:
         """Display the block graph in 3D."""
+        bytes_buffer = BytesIO()
         from tqec.sketchup.collada import (
             display_collada_model,
             write_block_graph_to_dae_file,
         )
 
-        bytes_buffer = BytesIO()
         write_block_graph_to_dae_file(self, bytes_buffer, pipe_length)
         return display_collada_model(
             filepath_or_bytes=bytes_buffer.getvalue(),
             write_html_filepath=write_html_filepath,
         )
+
+    def to_circuit(self, dimension: LinearFunction) -> cirq.Circuit:
+        """Build and return the quantum circuit representing the computation.
+
+        A block graph can be interpreted as a quantum computation.
+        Each cube represents a self contained layer of computations.
+        Each pipe modifies the initial and final layer of the cubes it connects.
+
+        Raises:
+            TQECException: if any of the circuits obtained by instantiating the
+                computational blocks is contains a qubit that is not a cirq.GridQubit.
+
+        Returns:
+            a cirq.Circuit instance representing the full computation.
+        """
+        blocks: dict[Position3D, StandardComputationBlock] = {}
+        for cube in self.cubes:
+            blocks[cube.position] = cube_to_block(cube, dimension)
+        for pipe in self.pipes:
+            u_computation, v_computation = (
+                blocks[pipe.u.position],
+                blocks[pipe.v.position],
+            )
+            u_computation.replace_boundary_plaquettes(pipe.direction)
```

Comment:
```suggestion
            u_computation.replace_boundary_plaquettes(pipe.direction, outgoing=True)
```
for symmetry with the line just below.